tests/functional/xhr/ajax-events.test.js

```diff
@@ -33,6 +33,45 @@ testDriver.test('capturing XHR ajax events', xhrBrowsers, function (t, browser,
   }
 })
 
+testDriver.test('capturing large payload of XHR ajax events', xhrBrowsers, function (t, browser, router) {
+  const ajaxPromise = router.expectAjaxEvents()
+  const rumPromise = router.expectRum()
+  const loadPromise = browser.safeGet(router.assetURL('xhr-large-payload.html', {
+    loader: 'spa',
+    init: {
+      ajax: {
+        harvestTimeSeconds: 5,
+        maxPayloadSize: 500
+      }
+    }
+  }))
+
+  let ajaxBundles = 0
+
+  function parseAjaxPromise(response) {
+    const {body, query} = response
+    const decoded = querypack.decode(body && body.length ? body : query.e)
+    const ajaxEvents = decoded.filter(e => e.type === 'ajax' && e.path === '/json')
+    if (ajaxEvents.length) ajaxBundles++
+    console.log('AJAX RESPONSE!', ajaxBundles)
+    if (ajaxBundles < 2) router.expectAjaxEvents().then(r => parseAjaxPromise(r))
+    else {
+      t.ok(ajaxBundles > 1, 'AJAX load is split into multiple payloads and sent')
+      t.end()
+    }
+  }
+
+  Promise.all([ajaxPromise, loadPromise, rumPromise])
```

Comment:
This will wait only for one ajax harvest; should it wait for multiple?

Reply:
it will wait for multiple i believe, in the `parseAjaxPromise` method